src/audio/kpb.c

```diff
@@ -467,6 +472,15 @@ static int kpb_copy(struct comp_dev *dev)
 	sink = (kpb->state == KPB_STATE_BUFFERING) ? kpb->rt_sink
 	       : kpb->cli_sink;
 
+	/* Pause selector copy during draining and/or draining on demand.
+	 * We keep selector in this "paused" state as long as draining
+	 * is going and later during direct copy to client's/host sink,
+	 * in so called "draining on demand" state. In order to rearm
+	 * detection algorithm kpb_reset is needed.
+	 */
+	if (kpb->state != KPB_STATE_BUFFERING)
+		kpb->rt_sink->sink->state = COMP_STATE_PAUSED;
```

Comment:
Should we not use the set_state() api here (since it checks for obvious errors in current state).

Reply:
@lgirdwood, I assume you meant *comp_set_state(struct comp_dev *dev, int cmd)*, in this case I think we can indeed use this interface.

Reply:
yes, I hand not had my coffee at that point :)

Reply:
@mrajwa @lgirdwood Instead of changing internal state of one component we could return `PPL_STATUS_PATH_STOP`. This way the copy won't propagate further to real time sink.

Reply:
@tlauda @lgirdwood , PR updated.